Python/sysmodule.c

```diff
@@ -2861,6 +2861,72 @@ _PySys_InitCore(PyThreadState *tstate, PyObject *sysdict)
     return _PyStatus_ERR("can't initialize sys module");
 }
 
+#ifdef HAVE_FORK
+static int
+stdio_at_fork_reinit(_Py_Identifier *key)
+{
+
+    int ret = 0;
+
+    PyObject *isatty = NULL;
+    PyObject *buffer = NULL;
+    PyObject *result = NULL;
+
+    PyObject *stdio = _PySys_GetObjectId(key);
+
+    _Py_IDENTIFIER(isatty);
+    isatty = _PyObject_CallMethodIdNoArgs(stdio, &PyId_isatty);
+    if (isatty == NULL) {
+        PyErr_Clear();
+        goto end;
+    }
+    if (Py_IsFalse(isatty)) {
+        goto end;
+    }
+
+    _Py_IDENTIFIER(buffer);
+    if (_PyObject_LookupAttrId(stdio, &PyId_buffer, &buffer) < 0) {
+        /* stdout.buffer and stderr.buffer are not part of the
+         * TextIOBase API and may not exist in some implementations.
+         * If not present, no need to reinitialize their locks. */
+        goto end;
+    }
+
+    _Py_IDENTIFIER(_at_fork_reinit);
+    if (_PyObject_GetAttrId(buffer, &PyId__at_fork_reinit) == NULL) {
```

Comment:
This calls PyObject_GetAttr, which returns a new reference, so you're leaking a reference here.

Reply:
Thanks for the tip! Should be fixed in the current version (along with the added tests and removal of the Python-accessible method.)